From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SoSeDiK <mrsosedik@gmail.com>
Date: Sat, 20 Sep 2025 15:54:47 +0300
Subject: [PATCH] Prevent clashed stew effects

One of the server advancements requires collecting each of crafted, traded, and looted suspicious stews.
They are differentiated by the effect duration. However, some of the looted tick durations clash with
the traded/crafted ones. This patch makes it so the looted stews don't generate with ticks that can clash.

diff --git a/net/minecraft/data/loot/packs/VanillaArchaeologyLoot.java b/net/minecraft/data/loot/packs/VanillaArchaeologyLoot.java
index 757baaf0da3ae615d3ee8d996bfe5ba7ce8c3b56..fdb6ef7f3afb14053a4ab761f7a3fd7d9126e0c5 100644
--- a/net/minecraft/data/loot/packs/VanillaArchaeologyLoot.java
+++ b/net/minecraft/data/loot/packs/VanillaArchaeologyLoot.java
@@ -33,11 +33,11 @@ public record VanillaArchaeologyLoot(HolderLookup.Provider registries) implement
                                 .apply(
                                     SetStewEffectFunction.stewEffect()
                                         .withEffect(MobEffects.NIGHT_VISION, UniformGenerator.between(7.0F, 10.0F))
-                                        .withEffect(MobEffects.JUMP_BOOST, UniformGenerator.between(7.0F, 10.0F))
-                                        .withEffect(MobEffects.WEAKNESS, UniformGenerator.between(6.0F, 8.0F))
-                                        .withEffect(MobEffects.BLINDNESS, UniformGenerator.between(5.0F, 7.0F))
-                                        .withEffect(MobEffects.POISON, UniformGenerator.between(10.0F, 20.0F))
-                                        .withEffect(MobEffects.SATURATION, UniformGenerator.between(7.0F, 10.0F))
+                                        .withEffect(MobEffects.JUMP_BOOST, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(7.0F, 10.0F), 8F)) // Kiterino - Prevent clashed stew effects
+                                        .withEffect(MobEffects.WEAKNESS, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(6.0F, 8.0F), 7F)) // Kiterino - Prevent clashed stew effects
+                                        .withEffect(MobEffects.BLINDNESS, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(5.0F, 7.0F), 6F)) // Kiterino - Prevent clashed stew effects
+                                        .withEffect(MobEffects.POISON, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(10.0F, 20.0F), 11F, 14F)) // Kiterino - Prevent clashed stew effects
+                                        .withEffect(MobEffects.SATURATION, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(7.0F, 10.0F), 7F)) // Kiterino - Prevent clashed stew effects
                                 )
                         )
                 )
diff --git a/net/minecraft/data/loot/packs/VanillaChestLoot.java b/net/minecraft/data/loot/packs/VanillaChestLoot.java
index da21f84d84f552e9287d3115c0ab2db1f6fa4fdd..b9004f48bbf6d0808e655b895dc4b4854d06608b 100644
--- a/net/minecraft/data/loot/packs/VanillaChestLoot.java
+++ b/net/minecraft/data/loot/packs/VanillaChestLoot.java
@@ -186,7 +186,7 @@ public record VanillaChestLoot(HolderLookup.Provider registries) implements Loot
                                 .apply(
                                     SetStewEffectFunction.stewEffect()
                                         .withEffect(MobEffects.NIGHT_VISION, UniformGenerator.between(7.0F, 10.0F))
-                                        .withEffect(MobEffects.BLINDNESS, UniformGenerator.between(5.0F, 7.0F))
+                                        .withEffect(MobEffects.BLINDNESS, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(5.0F, 7.0F), 6F)) // Kiterino - Prevent clashed stew effects
                                 )
                                 .apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 6.0F)))
                         )
@@ -1525,11 +1525,11 @@ public record VanillaChestLoot(HolderLookup.Provider registries) implements Loot
                             .apply(
                                 SetStewEffectFunction.stewEffect()
                                     .withEffect(MobEffects.NIGHT_VISION, UniformGenerator.between(7.0F, 10.0F))
-                                    .withEffect(MobEffects.JUMP_BOOST, UniformGenerator.between(7.0F, 10.0F))
-                                    .withEffect(MobEffects.WEAKNESS, UniformGenerator.between(6.0F, 8.0F))
-                                    .withEffect(MobEffects.BLINDNESS, UniformGenerator.between(5.0F, 7.0F))
-                                    .withEffect(MobEffects.POISON, UniformGenerator.between(10.0F, 20.0F))
-                                    .withEffect(MobEffects.SATURATION, UniformGenerator.between(7.0F, 10.0F))
+                                    .withEffect(MobEffects.JUMP_BOOST, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(7.0F, 10.0F), 8F)) // Kiterino - Prevent clashed stew effects
+                                    .withEffect(MobEffects.WEAKNESS, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(6.0F, 8.0F), 7F)) // Kiterino - Prevent clashed stew effects
+                                    .withEffect(MobEffects.BLINDNESS, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(5.0F, 7.0F), 6F)) // Kiterino - Prevent clashed stew effects
+                                    .withEffect(MobEffects.POISON, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(10.0F, 20.0F), 11F, 14F)) // Kiterino - Prevent clashed stew effects
+                                    .withEffect(MobEffects.SATURATION, new me.sosedik.kiterino.util.loot.OmittingUniformGenerator(UniformGenerator.between(7.0F, 10.0F), 7F)) // Kiterino - Prevent clashed stew effects
                             )
                     )
                     .add(LootItem.lootTableItem(Items.COAL).setWeight(6).apply(SetItemCountFunction.setCount(UniformGenerator.between(2.0F, 8.0F))))
